### PR TITLE
suppress target parameters - fixes #5

### LINKED ISSUE #5: Source and target parameters for I/O components should be suppressed
Source and target parameters for I/O components should be suppressed such that the Galaxy user cannot override them and access arbitrary files on the Galaxy host.

Likewise, PARAM_OVERWRITE should be suppressed to avoid users overwriting files on the Galaxy host.

### PR DESCRIPTION
**Pull request by rcffc**
- suppress target parameters
- created dropdown menus for language
    - configured conditionals for model variants for each language

### CODE REVIEW

File: dkpro-meta-example-galaxy/src/main/resources/templates/toolFormat.xml [hunk 1]
```diff
@@ -60,6 +59,17 @@
       paramDecls.each { param ->
         def paramType = typeMap[param.type] ?: 'text';
         def paramValue = ( paramValues.find { it.name == param.name }?.value ) ?: ''; 
+      	
+      	if (param.name.toLowerCase() == 'language') {
+        	paramType = 'select';
+        } else if (param.name.toLowerCase().contains('encoding')) {
```
Comment (reckart) [inline dkpro-meta-example-galaxy/src/main/resources/templates/toolFormat.xml]: Using a "x || y || z" would save some space / be more readable here.

File: dkpro-meta-example-galaxy/src/main/resources/templates/toolEngine.xml [hunk 2]
```diff
@@ -50,19 +50,80 @@
...
+          		langs.each { lang ->
+ </gsp:scriptlet>
+			    	<option value="${lang}">$lang</option>
+ <gsp:scriptlet>
+			     }
+           	 } 
+ </gsp:scriptlet>
+		  	</param>
+ <gsp:scriptlet>
+		  	if (param.name == 'language') {
```
Comment (reckart) [inline dkpro-meta-example-galaxy/src/main/resources/templates/toolEngine.xml]: There are three cases:
* a component has models - then this should be a dropdown based on the models
* a component has no models but declares language capabilities - then this dropdown should be based on the languages declared by the capabilities
* a component has a language parameter but not explicit information about which languages are supported - in this case, it should be a free text field.

File: dkpro-meta-example-galaxy/src/main/resources/templates/toolEngine.xml [hunk 3]
```diff
@@ -50,19 +50,80 @@
...
-    <gsp:scriptlet>
+            help="${html2text(param.description)}"> 
+ <gsp:scriptlet>
+            if (param.name.toLowerCase() == 'language') {
+            	def langs = engine.languages as Set
+          		engine.models.each { model -> 
+          			def lang = (model.@language.text())
+          			langs.add(lang)
+          			}
+          		langs.each { lang ->
```
Comment (reckart) [inline dkpro-meta-example-galaxy/src/main/resources/templates/toolEngine.xml]: Languages should be ordered.

Comment (rcffc) [reply dkpro-meta-example-galaxy/src/main/resources/templates/toolEngine.xml]: Languages are already ordered.

File: dkpro-meta-example-galaxy/src/main/resources/templates/toolEngine.xml (hunk 2 above)
Comment (rcffc) [reply dkpro-meta-example-galaxy/src/main/resources/templates/toolEngine.xml]: First two cases were handled, now the third case also applies.

**Comment by reckart** [comment]
test this please

**Comment by reckart** [comment]
@rcffc I'm testing a Jenkins plugin - please ignore my comments ;)

**Comment by reckart** [comment]
test this please

**Comment by reckart** [comment]
test this please

**Comment by reckart** [comment]
test this please

**Comment by reckart** [comment]
test this please

**Comment by reckart** [comment]
test this please

**Comment by reckart** [comment]
test this please

**Comment by reckart** [comment]
test this please

**Comment by reckart** [comment]
Jenkins, could you test this please?

**Pull request merged**